feat: add 2022 survey questions (#173)
nodejs/next-10: surveys/2022-12/questions.md

> # Survey December 2022

Reviewer: We should probably get this survey out soon 😄

Author: Haha yeah, I may have to update that :) 

a good timeline would be to 
- have it finished and published end of feb/start of march 
-  let it go a "month" until April/mid April
- review the feedback before Vancouver
- discuss about it during the Vancouver event

Reviewer 2: I think what we have in the doc makes sense. I think asking for help to do the survey from the Foundation makes senses sooner than later and we can agree on the schedule with them (assuming they can help). @joesepi I know you helped with some past surveys, who at the Foundation do you think we should reach out to for help on the survey?

> # Survey December 2022

Reviewer 2: ```suggestion
# Survey April 2023
```

    - Innovation at a consumable pace
    - Easy Installation
    - Easy issue reporting, resolution and collaboration
    - Broad deployment platform support
    - Broad desktop platform support
    - Consistent and intuitive error handling
    - Runtime diagnostic tooling
    - Development time diagnostic tooling
    - Relevant APIs in core
    - Module/dependency info and management
    - Ways to fund their work
    - Ability to assess impact of changes they make
    - Reasonable resource usage/performance
    - Good security and CVE practices in the project
    - Better CVE management in the ecosystem
    - Good CI infrastructure and experience in the project
    - Supportive Collaborators and Environment in the project
    - Better ways to build consensus in the project
    - Easy contribution workflow
    - Ability to embed and bundle the Node.js runtime
    - A well maintained and secure standard library
    - Assets that show Node.js is a good choice

- What is important to you that is not in this list? (open question)

## Technical Questions

- Are you using native ESM (e.g. `"type": "module"`,  `.mjs` or other solutions) in your Node.js Development?
    - Yes
    - No

> - Are you planning to convert your application to native ESM?

Reviewer: ```suggestion
- Are you planning to convert an application to native ESM or use native ESM for a future application?
```

At the June 2022 conference I polled the crowd with this question and almost no one was converting but almost everyone was planning to start their _next_ project as native ESM.

# Survey December 2022

## Personal question

- Which constituencies do you identify with?
    - Direct end users: Users who run tools themselves
    - Application operators: Users who interacts with existingm running applications
    - Application developers: Frontend tools consumers, backend server authors, tools authors
    - Library & package authors: Users who write librairies and packages to be included on other applications
    - Node.js core maintainers: Developers working directly on Node.js, individuals participating in Working Groups

>     - Organization with investments in Node.js (eg: Enterprises, Gouvernement bodies, Startups, Non-profit)

Reviewer 2: We should check if there were some that we added after the last survey, I think there was.

- Are you part of a constituency not covered ?
    - yes
    - no
- If not covered what are you part with ? (open question)
- Where are you from? (open question or countries list)
    - country
-  How long have you been working with Node.js? (numbers list)
    - 0 - 10+

> - In what kind of companies are you working in?

Author: -companies + organizations

    - big tech
    - company
    - startup
    - university

>     - other

Author: + individuals

- Are you part/following some working groups?
    - yes

>     - no

Reviewer 2: Possibly add "If so which ones?"

## Tech questions

- What is important for you ? (Multiple choices)
    - Good and up to date documentation
    - Tutorial and examples of core / complexe features
    - Internationalization of the documentation
    - Good understanding of the direction of the project
    - Predictable and stable release
    - Innovation and new features 
    - Consistent and intuitive error handling
    - Runtime diagnostic tooling
    - Relevant API in core
    - Good understanding and process for security and CVE in the ecosystem
    - Loader hook (to allow native typescript support)

> - What is your use case of Node.js ?

Author: add this link and a question about priorities

- Are you using native ESM, e.g. `"type": "module"` or `.mjs` in your Node.js Development?
- Are you planning to convert your application to native ESM?
- Any recurrent issue with Node.js ? (open question)

> - What need do you have not covered ? (open question)

Reviewer 2: we should add dnf and yum to the apt-get, brew list.

> - What need do you have not covered ? (open question)

Reviewer 3: oops, sorry, that should be `nvs`

    - Application operators: Users who interacts with existingm running applications
    - Application developers: Frontend tools consumers, backend server authors, tools authors
    - Library & package authors: Users who write librairies and packages to be included on other applications
    - Node.js core maintainers: Developers working directly on Node.js, individuals participating in Working Groups
    - Organization with investments in Node.js (eg: Enterprises, Gouvernement bodies, Startups, Non-profit, Education, Security)
- Are you part of a constituency not covered ?
    - yes
    - no
- If not covered what are you part with ? (open question)
- Where are you from? (open question or countries list)
    - country
-  How long have you been working with Node.js? (numbers list)
    - 0 - 10+
- In what kind of organizations are you working in?
    - big tech
    - company
    - startup
    - university
    - individual
    - other
- Are you part/following some Working Groups? 
    - yes
    - no
- If part/following some Working Groups, which one? (open question)

## Tech questions

- What is important for you ? (Multiple choices)
    - Good and up to date documentation

>     - Tutorial and examples of core / complexe features

Reviewer: ```suggestion
    - Tutorial and examples of core / complex features
```

    - Internationalization of the documentation
    - Good understanding of the direction of the project
    - Predictable and stable release
    - Innovation and new features 
    - Consistent and intuitive error handling
    - Runtime diagnostic tooling
    - Relevant API in core
    - Good understanding and process for security and CVE in the ecosystem

>     - Loader hook (to allow native typescript support)

Reviewer 4: ```suggestion
    - ESM Loader hook (to allow native typescript support)
```
also would this be "native", or merely equivalent to CJS's "register" patterns?

>     - Loader hook (to allow native typescript support)

Reviewer: ```suggestion
    - Hooks for customization, for example to use TypeScript without a separate build step
```

This isn’t specific to ESM.

Reviewer 4: but the capability already exists in CJS.

- What is your use case of Node.js ?
- Do you see others Technical Priorities ? [Technical Priorities](https://github.com/nodejs/node/blob/main/doc/contributing/technical-priorities.md) (open question)

> - Are you using native ESM, e.g. `"type": "module"` or `.mjs` in your Node.js Development?

Reviewer: I think the current wording is clearer than the suggestion.

> - Are you using native ESM, e.g. `"type": "module"` or `.mjs` in your Node.js Development?

Reviewer 4: "type module" is not automatically native ESM, though, and the confusion around that point has caused boundless newcomer confusion.

> - Are you using native ESM, e.g. `"type": "module"` or `.mjs` in your Node.js Development?

Author: I don't really have a great way to word that and I don't find the other wording clearer (nor have a better idea). Should this question be splited to differentiate that ?

- Are you planning to convert your application to native ESM?
- For those of you wishing to use ESM in an existing application, what have been the pain points or blockers preventing you from doing so (if any)?
- Any recurrent issue with Node.js ? (open question)
- What need do you have not covered ? (open question)
- How do you get your Node.js executables?
    - With a Node.js version manager: nvm, ns, n, etc.
    - Using the official installer

>     - Using package managers like npm or yarn and install Node.js as a package

Reviewer 5: I'm aware of a private package manager that installs the specified version of Node.js in the projects' settings automatically when it installs dependencies. It is similar to a `.nvmrc` local to a project.

    - Using a system package manager: apt-get, brew, dnf, yum ...
    - Other

> - What is your Architecture and OS in which you are running Node.js ? Do you plan to change that in the near future? (open question)

Reviewer 6: I think this question would be better as a multiple choice, using the same names from our supported OS/architecture list.

Reviewer: Also I would think that lots of people develop in one OS/arch and deploy in another, like develop in Mac and deploy under Linux/Docker.

Author: Add the question for both 
- deployed
- developer

Author: updated with the two questions. Is the list of architecture supported this one ? Or is there another file containing the list ? https://nodejs.org/api/process.html#processarch 

```
Possible values are: 'arm', 'arm64', 'ia32', 'mips','mipsel', 'ppc', 'ppc64', 's390', 's390x', and 'x64'.
```
